### PR TITLE
all done

### PR DESCRIPTION
**Pull request by vlykhvar**

### CODE REVIEW

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                "Jan", "Jan@net.ua"));
+        userService.add(new User("X Æ A-12",
+                "Musk", "musk@net.ua"));
+        return "inject";
+    }
+
+    @Override
+    @ResponseBody
+    @GetMapping("user/{id}")
+    public UserResponseDto get(@PathVariable("id") Long userId) {
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java:43]: @PathVariable("/{id}")

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return UserMapper.convert(userService.getById(userId));
+    }
+
+    @Override
+    @ResponseBody
+    @GetMapping("/user/")
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]: The better variant
```suggestion
    @GetMapping("/user")
```

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    public List<UserResponseDto> getAll() {
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]: and here add a path that you can get all at a new link

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.stream.Collectors;
+import models.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.ResponseBody;
+import service.UserMapper;
+import service.UserService;
+
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]:
```suggestion
@RestConroller
```

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+public class UserControllerImpl implements UserController {
+    private final UserService userService;
+
+    @Autowired
+    UserControllerImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    @ResponseBody
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]: remove it (@ResponseBody)

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                "Jora", "Jora@net.ua"));
+        userService.add(new User("Jan",
+                "Jan", "Jan@net.ua"));
+        userService.add(new User("X Æ A-12",
+                "Musk", "musk@net.ua"));
+        return "inject";
+    }
+
+    @Override
+    @ResponseBody
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]: remove it (@ResponseBody)

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    @GetMapping("user/{id}")
+    public UserResponseDto get(@PathVariable("id") Long userId) {
+        return UserMapper.convert(userService.getById(userId));
+    }
+
+    @Override
+    @ResponseBody
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]: remove it (@ResponseBody)

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    @GetMapping("/user/")
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers()
+                .stream()
+                .map(UserMapper::convert)
+                .collect(Collectors.toList());
+    }
+
+    @ResponseBody
```
Comment (gigbat) [inline src/main/java/controllers/impl/UserControllerImpl.java]: remove it (@ResponseBody)

**Review by gigbat — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/impl/UserControllerImpl.java (hunk 2 above)
Comment (vlykhvar) [reply src/main/java/controllers/impl/UserControllerImpl.java]: agree, done

**Review by gigbat — APPROVED** [review]

File: src/main/java/controllers/UserController.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package controllers;
+
+import dto.UserResponseDto;
+import java.util.List;
+
+public interface UserController {
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/UserController.java]: there is no need to create interfaces for controller layer - please remove it

File: src/main/java/config/WebConfig.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {"controllers.impl"})
```
Comment (KseniiaMakarova) [inline src/main/java/config/WebConfig.java]:
```suggestion
@ComponentScan(basePackages = {"controllers"})
```

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import service.UserMapper;
+import service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserControllerImpl implements UserController {
+    private final UserService userService;
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/impl/UserControllerImpl.java]: UserMapper should also be a dependency here - don't make the methods static there

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Autowired
+    UserControllerImpl(UserService userService) {
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/impl/UserControllerImpl.java]: is anything missing?

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+                "Jora", "Jora@net.ua"));
+        userService.add(new User("Jan",
+                "Jan", "Jan@net.ua"));
+        userService.add(new User("X Æ A-12",
+                "Musk", "musk@net.ua"));
+        return "inject";
+    }
+
+    @Override
+    @GetMapping("{id}")
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/impl/UserControllerImpl.java]: does this work?

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    public UserResponseDto get(@PathVariable Long id) {
+        return UserMapper.convert(userService.getById(id));
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/impl/UserControllerImpl.java]:
```suggestion
        return userMapper.mapToDto(userService.getById(id));
```

File: src/main/java/controllers/impl/UserControllerImpl.java [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    @GetMapping("/get")
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/impl/UserControllerImpl.java]: please see checklist
we should not use crud method names in mapping

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 16]
```diff
@@ -50,4 +51,12 @@ public void add(User user) {
             throw new RuntimeException("Error getting list of users", e);
         }
     }
+
+    public Optional<User> get(Long id) {
```
Comment (KseniiaMakarova) [inline src/main/java/dao/impl/UserDaoImpl.java:56]:
```suggestion
    @Override
    public Optional<User> get(Long id) {
```

File: src/main/java/service/UserMapper.java [hunk 17]
```diff
@@ -0,0 +1,16 @@
+package service;
+
+import dto.UserResponseDto;
+import models.User;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
+    public static UserResponseDto convert(User user) {
```
Comment (KseniiaMakarova) [inline src/main/java/service/UserMapper.java]:
```suggestion
    public UserResponseDto mapToDto(User user) {
```

File: src/main/java/service/impl/UserServiceImpl.java [hunk 18]
```diff
@@ -25,4 +25,10 @@ public void add(User user) {
     public List<User> listUsers() {
         return userDao.listUsers();
     }
+
+    @Override
+    public User getById(Long id) {
+        return userDao.get(id).orElseThrow(()
+                -> new RuntimeException("Can't fing user by id" + id));
```
Comment (KseniiaMakarova) [inline src/main/java/service/impl/UserServiceImpl.java]:
```suggestion
                -> new RuntimeException("Can't find user by id" + id));
```

File: src/main/java/config/WebConfig.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+public class WebConfig {
+    @Bean
+    public InternalResourceViewResolver resolver() {
+        InternalResourceViewResolver resolver = new InternalResourceViewResolver();
+        resolver.setViewClass(JstlView.class);
+        resolver.setPrefix("/WEB-INF/views/");
+        resolver.setSuffix(".jsp");
+        return resolver;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/config/WebConfig.java]: will this work without this bean and jsp dependencies in pom.xml?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/impl/UserControllerImpl.java (hunk 13 above)
Comment (vlykhvar) [reply src/main/java/controllers/impl/UserControllerImpl.java]: yep

File: src/main/java/controllers/impl/UserControllerImpl.java (hunk 15 above)
Comment (vlykhvar) [reply src/main/java/controllers/impl/UserControllerImpl.java]: change

File: src/main/java/controllers/UserController.java [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+                "Musk", "musk@net.ua"));
+        return "inject";
+    }
+
+    @GetMapping("{id}")
+    public UserResponseDto get(@PathVariable Long id) {
+        return userMapper.mapToDto(userService.getById(id));
+    }
+
+    @GetMapping("/all")
```
Comment (KseniiaMakarova) [inline src/main/java/controllers/UserController.java]:
```suggestion
    @GetMapping
```
the correct mapping for `getAll` operation will be just `/user/`
please see task description

File: src/main/java/service/impl/UserServiceImpl.java (hunk 18 above)
Comment (KseniiaMakarova) [inline src/main/java/service/impl/UserServiceImpl.java]: what about my previous comment on the spelling mistake here?

File: src/main/java/config/WebConfig.java [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+@ComponentScan(basePackages = {"controllers"})
+public class WebConfig {
+    @Bean
+    public InternalResourceViewResolver resolver() {
+        InternalResourceViewResolver resolver = new InternalResourceViewResolver();
+        resolver.setViewClass(JstlView.class);
+        resolver.setPrefix("/WEB-INF/views/");
+        resolver.setSuffix(".jsp");
+        return resolver;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/config/WebConfig.java]: same here  - have you tried removing this?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**